### PR TITLE
ME-1976: e2e encryption for mysql, connector org certifcate and system CA pool

### PR DESCRIPTION
**Pull request by btoonk**
# Description

* e2e encryption support for mysql
* use connector org certificate for e2e encryption
* use system CA pool for verifying the public certs

Fixes # (ME-1976)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tested in staging.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Review by adrianosela — CHANGES_REQUESTED** [review]
If we know what the proxy server cert is, then let's not accept `x509.SystemCertPool` -- let's build our own e.g.:

```
	pemBytes, err := os.ReadFile(trustedCertificate)
	if err != nil {
		log.Fatalf("failed to read trusted cert file: %v", err)
	}

	der, _ := pem.Decode(pemBytes)
	cert, err := x509.ParseCertificate(der.Bytes)
	if err != nil {
		log.Fatalf("failed to parse certificate: %v", err)
	}

	cp := x509.NewCertPool()
	cp.AddCert(cert)

	conn, err := tls.Dial("tcp", ":2424", &tls.Config{
		RootCAs:    cp,
		ServerName: "localhost",
	})
	if err != nil {
		log.Fatalf("failed to dial TLS: %v", err)
	}
```

(https://github.com/adrianosela/kmsca/blob/main/__example__/4_demos/tls_server/client/main.go#L15-L32)

**Comment by adrianosela** [comment]
Rebased onto main

**Comment by btoonk** [comment]
> If we know what the proxy server cert is, then let's not accept `x509.SystemCertPool` -- let's build our own e.g.:
> 
> ```
> 	pemBytes, err := os.ReadFile(trustedCertificate)
> 	if err != nil {
> 		log.Fatalf("failed to read trusted cert file: %v", err)
> 	}
> 
> 	der, _ := pem.Decode(pemBytes)
> 	cert, err := x509.ParseCertificate(der.Bytes)
> 	if err != nil {
> 		log.Fatalf("failed to parse certificate: %v", err)
> 	}
> 
> 	cp := x509.NewCertPool()
> 	cp.AddCert(cert)
> 
> 	conn, err := tls.Dial("tcp", ":2424", &tls.Config{
> 		RootCAs:    cp,
> 		ServerName: "localhost",
> 	})
> 	if err != nil {
> 		log.Fatalf("failed to dial TLS: %v", err)
> 	}
> ```
> 
> (https://github.com/adrianosela/kmsca/blob/main/__example__/4_demos/tls_server/client/main.go#L15-L32)

We already rely on the system pool for the connection to the API.
We do know it's a Let's encrypt certificate, but I don't want to hardcode those intermediates (or root).
Why do you think using the system pool is bad?

**Comment by adrianosela** [comment]
> > If we know what the proxy server cert is, then let's not accept `x509.SystemCertPool` -- let's build our own e.g.:
> > ```
> > 	pemBytes, err := os.ReadFile(trustedCertificate)
> > 	if err != nil {
> > 		log.Fatalf("failed to read trusted cert file: %v", err)
> > 	}
> > 
> > 	der, _ := pem.Decode(pemBytes)
> > 	cert, err := x509.ParseCertificate(der.Bytes)
> > 	if err != nil {
> > 		log.Fatalf("failed to parse certificate: %v", err)
> > 	}
> > 
> > 	cp := x509.NewCertPool()
> > 	cp.AddCert(cert)
> > 
> > 	conn, err := tls.Dial("tcp", ":2424", &tls.Config{
> > 		RootCAs:    cp,
> > 		ServerName: "localhost",
> > 	})
> > 	if err != nil {
> > 		log.Fatalf("failed to dial TLS: %v", err)
> > 	}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > (https://github.com/adrianosela/kmsca/blob/main/__example__/4_demos/tls_server/client/main.go#L15-L32)
> 
> We already rely on the system pool for the connection to the API. We do know it's a Let's encrypt certificate, but I don't want to hardcode those intermediates (or root). Why do you think using the system pool is bad?

I misunderstood what these certs were. Ignore my previous comment.

File: internal/sqlauthproxy/mysql_local_handler.go [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+	serverConn      *server.Conn
+	statements      map[int64]*client.Stmt
+	preparedQueries map[int64]string
+	lastAuth        time.Time
+	recordingChan   chan message
+}
+
+var _ server.Handler = &mysqlLocalHandler{}
+
+func (h *mysqlLocalHandler) Database() string {
```
Comment (adrianosela) [inline internal/sqlauthproxy/mysql_local_handler.go:41]: godoc all over this file. You can copy the explanations from the public interface.

**Review by adrianosela — APPROVED** [review]
🚢 it. Next one try splitting the work into small thin slices. Hard to review such a large PR.

**Pull request merged**